src/datasource.ts

```diff
@@ -49,12 +50,21 @@ export default class VividCortexDatasource {
 
   metricFindQuery(query: string) {
     const params = {
-      q: this.interpolateVariables(query),
+      from: moment()
+        .utc()
+        .subtract(7, 'days')
+        .unix(),
+      until: moment()
+        .utc()
+        .unix(),
+      new: '0',
+      filter: query ? `*${query}*` : undefined,
     };
 
-    return this.doRequest('metrics/search', 'GET', params)
+    return this.doRequest('metrics', 'GET', params)
```

Comment:
@mtrias I first started using the `/metrics` endpoint instead of `/metrics/search`, which I thought it required a `host` parameter to work. This required an extra request (2n in the beginning, n+1 after some modifications), which ended up causing problems in some edge cases, depending on when the controller was instantiated (just context, no need to know more about this).

Then, I realized I could use this endpoint without the `host` parameters and `/hosts` call (removing the need to have 2n or n+1 XHR requests), which results in the implementation you're looking at here. So, here comes the question: **is it alright to use it without specifying the hosts? or is it required?**.

I left the old implementation [here](https://github.com/VividCortex/grafana-datasource/pull/24/commits/3c773df160b2023abbf125d9c5ebe6421d523328) for reference.